### DISCUSSION TITLE
`vercel dev` throws `Error: Watcher error: Error: EMFILE: too many open files, watch` on gatsby project

### DISCUSSION THREAD

**Discussion by chandlervdw** [vercel/vercel#8936] [Help]
`gatsby develop` works fine but when running `vercel dev`, I get the following error:
```
Error: Watcher error: Error: EMFILE: too many open files, watch
```
and the process hangs.

When I run `vercel dev -d`, it seems that `@vercel/node` may be the culprit:
```
> [debug] [2022-11-17T17:43:41.159Z] Imported Builder "@vercel/node" from "/Users/chandler/.config/yarn/global/node_modules/@vercel/node"
> [debug] [2022-11-17T17:43:41.177Z] Adding build match for "middleware.ts" with "@vercel/node@latest"
Error: Watcher error: Error: EMFILE: too many open files, watch
> [debug] [2022-11-17T17:43:42.597Z] Closing file watcher
```

I have found issues around the web concerning similar node errors but none concerning vercel or gatsby other than [this thread](https://github.com/vercel/vercel/discussions/7856#discussioncomment-3002080). Installing `watchman` via homebrew (`brew install watchman`) and reinstalling `node_modules` did not help.

**Comment by amyegan** [comment]
Thanks for reaching out @chandlervdw! I was not able to produce this error with the [Gatsby Boilerplate](https://vercel.com/templates/gatsby/gatsbyjs-boilerplate) template. Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)?
